### PR TITLE
MTA_2174 insert mta-cli attribute instead of plain text windup-cli

### PR DESCRIPTION
**Pull request by emarcusRH**
https://issues.redhat.com/browse/MTA-2174

Replace "windup-cli" plain text with attribute {mta-cli}, 
which resolves to "mta-cli" 

Changes in the following files: 
_docs/topics/vs-code-extension-run-configuration.adoc_  REMOVED due to merge conflict
_docs/topics/using-openrewrite-recipes.adoc_
_docs/topics/cli-run.adoc_
_docs/topics/review-reports.adoc_
_docs/topics/mavenize.adoc_
_docs/topics/manually-test-rules.adoc_
_docs/topics/intellij-idea-plugin-run-configuration.adoc_
_docs/topics/create-first-xml-rule.adoc_  REMOVED due to merge conflict

Previews: 
https://deploy-preview-828--windup-documentation.netlify.app/docs/cli-guide/master/
https://deploy-preview-828--windup-documentation.netlify.app/docs/rules-development-guide/master/
https://deploy-preview-828--windup-documentation.netlify.app/docs/vs-code-extension-guide/master/
https://deploy-preview-828--windup-documentation.netlify.app/docs/rules-development-guide/master/
https://deploy-preview-828--windup-documentation.netlify.app/docs/intellij-idea-plugin-guide/master/

### CODE REVIEW

**Comment by emarcusRH** [comment]
I pushed one additional commit to update the _document-attributes.adoc_
placing the _mtr-cli_ attribute inside the ifdef sections for **MTA** and **MTR**, so that the attribute is now resolved correctly for both documentation sets.

File: docs/topics/cli-run.adoc [hunk 1]
```diff
@@ -11,11 +11,11 @@ You can run {ProductShortName} against your application.
 .Procedure
 
 . Open a terminal and navigate to the `<{ProductShortName}_HOME>/bin/` directory.
-. Execute the `windup-cli` script, or `windup-cli.bat` for Windows, and specify the appropriate arguments:
+. Execute the `{mta-cli}` script, or `{mta-cli}.bat` for Windows, and specify the appropriate arguments:
```
Comment (abrugaro) [inline docs/topics/cli-run.adoc:14]: mta-cli for windows is not a `.bat` anymore, now it is a `.exe` file

File: docs/topics/cli-run.adoc [hunk 2]
```diff
@@ -76,7 +76,7 @@ To override the default _Fernflower_ decompiler, pass the `-Dwindup.decompiler`
 
 [source,terminal,subs="attributes+"]
 ----
-$ <{ProductShortName}_HOME>/bin/windup-cli -Dwindup.decompiler=procyon \
+$ <{ProductShortName}_HOME>/bin/{mta-cli} -Dwindup.decompiler=procyon \
```
Comment (abrugaro) [inline docs/topics/cli-run.adoc:79]: The -D option doesn't exist anymore in mta-cli

File: docs/topics/using-openrewrite-recipes.adoc [hunk 3]
```diff
@@ -49,11 +49,11 @@ You can include more than one recipe by specifying each in the `activeRecipes` p
 ** `dryRun` : The script returns a list of proposed changes. Ignore the `"Run 'mvn rewrite:run' to apply the recipes"` message.
 ** `run`: The script applies the changes.
 
-. Run `windup-cli` with `--goal run` to apply the recipe:
+. Run `{mta-cli}` with `--goal run` to apply the recipe:
 +
 [source,terminal,subs="attributes+"]
 ----
-$ ./windup-cli --openrewrite --input </path/to/source/project> \
+$ ./{mta-cli} --openrewrite --input </path/to/source/project> \
```
Comment (abrugaro) [inline docs/topics/using-openrewrite-recipes.adoc:56]: The `--openrewrite` argument doesn't exist anymore. For code transformation the command is now:

```bash
mta-cli transform openrewrite ...
```

**Review by abrugaro — APPROVED** [review]
The main point of this PR LGTM.


I left some comments that should be addressed in this, or following PRs.

Apart from that I have two more consideration

1. All mta-cli analysis commands should start now with `mta-cli analyze` instead of just `mta-cli`

2. The available openrewrite rules table has hard-coded references to MTR

image

File: docs/topics/cli-run.adoc (hunk 2 above)
Comment (emarcusRH) [reply docs/topics/cli-run.adoc:79]: @abrugaro  what should this line look like? 
1.  $ <{ProductShortName}_HOME>/bin/{mta-cli} -windup.decompiler=procyon \
2.  $ <{ProductShortName}_HOME>/bin/{mta-cli}  \
3.  - remove this line completely?

Comment (abrugaro) [reply docs/topics/cli-run.adoc:79]: We should eliminate the entire section that refers to overriding the default decompiler

File: docs/topics/cli-run.adoc (hunk 1 above)
Comment (emarcusRH) [reply docs/topics/cli-run.adoc:14]: @abrugaro  does this apply to all mentions of batch files that I find in the documentation? 

I found these instances: 
NOTE: In a Windows environment, use the `run_mta.bat` script.

NOTE: In a Windows environment, use the `switch_to_authentication_required.bat` script.

C:\<{ProductShortName}_HOME>\run_windup.bat   (should this also say run_mta.exe???)

Execute the `{LC_PSN}-cli` script, or `{LC_PSN}-cli.bat` for Windows, and specify the appropriate arguments:

> ...\bin\standalone.bat -Djboss.socket.binding.port-offset=<offset_value>


C:\<MTR_HOME>\switch_to_authentication_required.bat

C:\<MTR_HOME>\run_windup.bat

C:\<MTR_HOME>\switch_to_automatic_authentication.bat

**Comment by emarcusRH** [comment]
> The main point of this PR LGTM.
> 
> I left some comments that should be addressed in this, or following PRs.
> 
> Apart from that I have two more consideration
> 
>     1. All mta-cli analysis commands should start now with `mta-cli analyze` instead of just `mta-cli`
> 
@abrugaro    I created a separate PR to fix this: 
https://github.com/windup/windup-documentation/pull/837

**Comment by emarcusRH** [comment]
these changes have been handled in a different PR - https://github.com/windup/windup-documentation/pull/847

**Pull request closed without merging**